### ISSUE TITLE
Side by Side Rendering

### ISSUE THREAD

**Issue by mlucool** [jupyterlab/jupyterlab#10207]
### Problem

Sometimes you have the horizonal space and want to see input/output side by side instead of top/bottom. This is the same issue as notebook's https://github.com/jupyter/notebook/issues/2897.

### Proposed Solution

Add a setting that allows a user to opt-into this layout mode.

This image is from https://github.com/jupyter/notebook/issues/2897, showing what it could look like:
![](https://user-images.githubusercontent.com/465582/31146291-78e56240-a886-11e7-96e3-a3fe069025c7.png)

**Comment by echarles** [comment]
Can simple CSS do the job, or do we need something more evolved to implement this?

**Comment by echarles** [comment]
There is a relation and impact with the output collapsing and the toc.

cc/ @jess-x

**Comment by jess-x** [comment]
@echarles I agree output collapsing and toc will have impact, and also I have seen some interesting codeMirror behaviors under different CSS display modes. (i.e I have seen it turns uneditable when the input-area wrapper is displayed as inline-flex)

**Comment by jess-x** [comment]
> I agree output collapsing and toc will have impact, and also I have seen some interesting codeMirror behaviors under different CSS display modes. (i.e I have seen it turns uneditable when the input-area wrapper is displayed as inline-flex)

That being said, tweaking some CSS and cherry-picking certain outputs does give us this satisfying looking proof-of-concept :) 

https://user-images.githubusercontent.com/22648112/125579268-62399392-0d40-46fb-aebf-64fea54b5805.mov

**Comment by echarles** [comment]
@jess-x That is promising. I wonder what it gives when you have a notebook with a mix of cells having outputs, and other without outputs. I have super quickly changed the css to get the outputs to the right, and the rendering of those mixed cells (with and without output) was not satisfying. Also, I wonder what it the behavior on input/output collapsing.

**Comment by jess-x** [comment]
@echarles I tried tweaking a few more css properties and this is what we have so far. This is definitely not a robust solution, but I hope it addresses some of the use cases :)

https://user-images.githubusercontent.com/22648112/125897907-fe53f132-9236-41cb-8ded-69b68c5d2a82.mov

**Comment by echarles** [comment]
Yeah, the cells width is not uniform. Have you investigated the usage of calc() https://developer.mozilla.org/en-US/docs/Web/CSS/calc() to fix the width of the cells? Try maybe to force the width to be the half of the full width with the calc() function.

**Comment by joelostblom** [comment]
I think this can be closed now that #10648 was merged?

**Comment by josselex** [comment]
Can this be configured in the jupyterlab config to be the default? I did not find anything about this.

**Comment by mlucool** [comment]
Yes. in 3.3.x (you can check out the latest rc), this is possible.
